From 30f0fc156bbb03d4237e11836fab76a2b385fee8 Mon Sep 17 00:00:00 2001
From: Rob Norris <rob.norris@truenas.com>
Date: Mon, 19 Oct 2026 08:25:28 +0000
Subject: [PATCH 1/4] Linux: use cpu_feature_enabled() instead of
 static_cpu_has()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

static_cpu_has() was renamed in 7.3 because since 3.18
cpu_feature_enabled() has always been the "right" way to do the same
thing. So we just switch everything over.

This is a verbatim rename of upstream commit 315bc75af, regenerated
against zfs-2.4.5-staging because upstream's hunks do not apply to the
context this branch carries.

(cherry picked from commit 315bc75afc18f548df3b738b69bfaad7cff094ad)
Signed-off-by: Niccolò Belli <niccolo.belli@linuxsystems.it>
---
 include/os/linux/kernel/linux/simd_x86.h | 14 +++++++-------
 module/zcommon/simd_stat.c               |  8 ++++----
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/include/os/linux/kernel/linux/simd_x86.h b/include/os/linux/kernel/linux/simd_x86.h
index 326f471d7..1d7890f15 100644
--- a/include/os/linux/kernel/linux/simd_x86.h
+++ b/include/os/linux/kernel/linux/simd_x86.h
@@ -315,24 +315,24 @@ kfpu_begin(void)
 	 */
 	uint8_t *state = zfs_kfpu_fpregs[smp_processor_id()];
 #if defined(HAVE_XSAVES)
-	if (static_cpu_has(X86_FEATURE_XSAVES)) {
+	if (cpu_feature_enabled(X86_FEATURE_XSAVES)) {
 		kfpu_do_xsave("xsaves", state, ~XFEATURE_MASK_XTILE);
 		return;
 	}
 #endif
 #if defined(HAVE_XSAVEOPT)
-	if (static_cpu_has(X86_FEATURE_XSAVEOPT)) {
+	if (cpu_feature_enabled(X86_FEATURE_XSAVEOPT)) {
 		kfpu_do_xsave("xsaveopt", state, ~XFEATURE_MASK_XTILE);
 		return;
 	}
 #endif
 #if defined(HAVE_XSAVE)
-	if (static_cpu_has(X86_FEATURE_XSAVE)) {
+	if (cpu_feature_enabled(X86_FEATURE_XSAVE)) {
 		kfpu_do_xsave("xsave", state, ~XFEATURE_MASK_XTILE);
 		return;
 	}
 #endif
-	if (static_cpu_has(X86_FEATURE_FXSR)) {
+	if (cpu_feature_enabled(X86_FEATURE_FXSR)) {
 		kfpu_save_fxsr(state);
 	} else {
 		kfpu_save_fsave(state);
@@ -380,18 +380,18 @@ kfpu_end(void)
 {
 	uint8_t  *state = zfs_kfpu_fpregs[smp_processor_id()];
 #if defined(HAVE_XSAVES)
-	if (static_cpu_has(X86_FEATURE_XSAVES)) {
+	if (cpu_feature_enabled(X86_FEATURE_XSAVES)) {
 		kfpu_do_xrstor("xrstors", state, ~XFEATURE_MASK_XTILE);
 		goto out;
 	}
 #endif
 #if defined(HAVE_XSAVE)
-	if (static_cpu_has(X86_FEATURE_XSAVE)) {
+	if (cpu_feature_enabled(X86_FEATURE_XSAVE)) {
 		kfpu_do_xrstor("xrstor", state, ~XFEATURE_MASK_XTILE);
 		goto out;
 	}
 #endif
-	if (static_cpu_has(X86_FEATURE_FXSR)) {
+	if (cpu_feature_enabled(X86_FEATURE_FXSR)) {
 		kfpu_restore_fxsr(state);
 	} else {
 		kfpu_restore_fsave(state);
diff --git a/module/zcommon/simd_stat.c b/module/zcommon/simd_stat.c
index 007ae9e4f..1391064ca 100644
--- a/module/zcommon/simd_stat.c
+++ b/module/zcommon/simd_stat.c
@@ -126,13 +126,13 @@ simd_stat_kstat_data(char *buf, size_t size, void *data)
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
 		    "osxsave", boot_cpu_has(X86_FEATURE_OSXSAVE));
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
-		    "xsaves", static_cpu_has(X86_FEATURE_XSAVES));
+		    "xsaves", cpu_feature_enabled(X86_FEATURE_XSAVES));
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
-		    "xsaveopt", static_cpu_has(X86_FEATURE_XSAVEOPT));
+		    "xsaveopt", cpu_feature_enabled(X86_FEATURE_XSAVEOPT));
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
-		    "xsave", static_cpu_has(X86_FEATURE_XSAVE));
+		    "xsave", cpu_feature_enabled(X86_FEATURE_XSAVE));
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
-		    "fxsr", static_cpu_has(X86_FEATURE_FXSR));
+		    "fxsr", cpu_feature_enabled(X86_FEATURE_FXSR));
 #endif /* __x86__ */
 #if defined(__arm__) || defined(__aarch64__)
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
-- 
2.55.0

